sources/cassandra/main.tf

```diff
@@ -0,0 +1,102 @@
+resource "google_compute_firewall" "allow_ports" {
+  name    = "allow-zdm-ports"
+  network = var.common_params.network
+
+  allow {
+    protocol = "tcp"
+    ports    = ["14002", "9042"]
+  }
+
+  source_ranges = ["0.0.0.0/0"]
+  target_tags   = ["zdm"]
+}
+
+resource "google_compute_instance" "cos_instances" {
+  count        = var.common_params.instance_count
+  name         = "${var.common_params.instance_name_prefix}-${count.index}"
+  machine_type = var.common_params.machine_type
+  zone         = var.common_params.zone
+
+  boot_disk {
+    initialize_params {
+      image = "projects/cos-cloud/global/images/cos-105-17412-535-55"
+      size  = 32
+      type  = "pd-balanced"
+    }
+  }
+
+  network_interface {
+    network = var.common_params.network
+    access_config {}
+  }
+
+  tags = ["zdm"]
+
+  metadata = {
+    enable-oslogin = "TRUE"
```

Comment:
Do we need `osLogin` to be forced as enable? If it's only for ease of  debugging, could we please take this in as a variable?

There could be also be configured by the customer's policy at project or org level
https://cloud.google.com/compute/docs/oslogin/manage-oslogin-in-an-org

The VM IIRC, overrides the project or org config and may be that's not the security posture the customer wants.

Reply:
Agree that production setups should have this disabled. But I do not expect customers to use this directly for production (just like our other terraform scripts) and this is primarily for reference and basic testing.

Adding a flag for this niche detail is a little out of place, as there are many other things customers would like to tweak in their GCE vm config, firewall, networking settings etc itself. This is just a reference on how to get the proxy stuff orchestrated.

Reply:
```
Adding a flag for this niche detail is a little out of place
```
Agreed, but my slight worry was that:
If we are overriding it enable at VM level, we might end up contradicting the policy that the customer's place has setup at a project or org level.
Should we just let it inherit the project instead?

If this is primarily for reference - 
  1. please let us call it out as such in the documentation
  2. if it's primarily for reference I might lean on not enabling OS login here and letting it inheirt from the project properties. We can always advise the customer to change the script if they want to override this in the docs.